airbyte-integrations/connectors/destination-langchain/destination_langchain/processor.py

```diff
@@ -0,0 +1,95 @@
+from typing import List, Optional, Tuple, Mapping
+from destination_langchain.config import ProcessingConfigModel
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from airbyte_cdk.models import AirbyteRecordMessage, ConfiguredAirbyteCatalog, ConfiguredAirbyteStream
+from langchain.document_loaders.base import Document
+from langchain.utils import stringify_dict
+import dpath.util
+from dpath.exceptions import PathNotFound
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+import uuid
+from dataclasses import dataclass
+
+METADATA_STREAM_FIELD = "_airbyte_stream"
+METADATA_NATURAL_ID_FIELD = "_natural_id"
+
+class Processor:
+    streams: Mapping[str, ConfiguredAirbyteStream]
+
+    def __init__(self, config: ProcessingConfigModel, catalog: ConfiguredAirbyteCatalog, max_metadata_size: Optional[int] = None):
+        self.streams = {stream.stream.name: stream for stream in catalog.streams}
+        self.max_metadata_size = max_metadata_size
+
+        self.splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
+            chunk_size=config.chunk_size, chunk_overlap=config.chunk_overlap
+        )
+        self.text_fields = config.text_fields
+
+    def process(self, record: AirbyteRecordMessage) -> Tuple[List[Document], Optional[str]]:
+        """
+        Generate documents from records.
+        :param records: List of AirbyteRecordMessages
+        :return: Tuple of (List of document chunks, Natural id to delete if applicable)
+        """
+        doc = self._generate_document(record)
+        chunks = self._split_document(doc)
+        id_to_delete = doc.metadata[METADATA_NATURAL_ID_FIELD] if METADATA_NATURAL_ID_FIELD in doc.metadata else None
+        return chunks, id_to_delete
+
+    def _generate_document(self, record: AirbyteRecordMessage) -> Document:
+        relevant_fields = self._extract_relevant_fields(record)
+        metadata = self._extract_metadata(record)
+        text = stringify_dict(relevant_fields)
+        return Document(page_content=text, metadata=metadata)
+
+    def _extract_relevant_fields(self, record: AirbyteRecordMessage) -> dict:
+        relevant_fields = {}
+        if self.text_fields:
+            for field in self.text_fields:
+                values = dpath.util.values(record.data, field, separator=".")
+                if values and len(values) > 0:
+                    relevant_fields[field] = values
+        else:
+            relevant_fields = record.data
+        return relevant_fields
+
+    def _extract_metadata(self, record: AirbyteRecordMessage) -> dict:
```

Comment:
if i'm reading this correctly metadata is everything in the input record that isn't a text/data field?

Reply:
right, we could also make it configurable. Technically you could use the column selection feature for this.